src/libraries/System.Net.Http/src/System/Net/Http/HttpMessageInvoker.cs

```diff
@@ -103,6 +101,12 @@ static async Task<HttpResponseMessage> SendAsyncWithTelemetry(HttpMessageHandler
             }
         }
 
+        private static bool ShouldSendWithTelemetry(HttpRequestMessage request) =>
+            HttpTelemetry.Log.IsEnabled() &&
+            !request.WasSentByHttpClient() &&
+            request.RequestUri is Uri requestUri &&
+            requestUri.IsAbsoluteUri;
```

Comment:
The call to `RequestStart` that follows this check will access Uri fields that would throw for relative Uris.

Adding this check here so that you get the meaningful exception message from `SocketsHttpHandler` instead.